projects/driver_controls/inc/event_arbiter.h

```diff
@@ -8,30 +8,34 @@
 // not pressed, which would be dangerous for the driver.
 
 #include "fsm.h"
+#include "objpool.h"
 #include "status.h"
 
 // Arbitrary FSM cap
 #define EVENT_ARBITER_MAX_FSMS 10
 
-typedef struct EventArbiterOutputData {
-  uint8_t id;
-  uint8_t state;
-  uint16_t data;
-} EventArbiterOutputData;
-
-// Typedef for the FSM arbitration functions
+// Returns whether the given event should be processed by any FSMs
 typedef bool (*EventArbiterCheck)(const Event *e);
 
-typedef void (*EventArbiterOutput)(EventArbiterOutputData data);
+typedef struct EventArbiter {
+  FSM *fsm;
+  EventArbiterCheck event_check_fn;
+} EventArbiter;
+
+typedef struct EventArbiterStorage {
```

Comment:
Couldn't this be kept private? I don't know if this would see much use outside of the event arbiter source code

Reply:
If we didn't we wouldn't be able to allocate it anywhere

Reply:
Couldn't it be allocated as a static variable? And then initialized when `event_arbiter_init` is called?

Reply:
Then you have a static variable for no real reason

Reply:
Sure, but even then wouldn't only one `EventArbiterStorage` struct to be allocated in order to keep track of the multiple arbiters? Additionally, its fields don't need to be accessed outside of event_arbiter.c.

Reply:
No, I mean multiple independent instances running in parallel. We never expose our structs so that people can access their fields. We expose them so that users can choose where to allocate their storage. If they wanted to make it heap-allocated, they can. If they wanted to allocate it as part of a larger struct, that's up to them. We could easily change this so that the user passes in the storage themselves instead of a fixed number.